src/document_change_manager.ts

```diff
@@ -405,16 +461,18 @@ export class DocumentChangeManager implements Disposable {
 
         await actions.lua("handle_changes", bufId, changeArgs);
 
+        if (!isDirty) {
+            await this.client.request("nvim_buf_set_option", [bufId, "modified", false]);
```

Comment:
Why is this line also in 412?

Reply:
The following events may occur:

1. The version remains unchanged, `contentChanges` is empty, and `isDirty` is false (the logic will not reach line 465, so `modified` needs to be set).
2. The version changes, `contentChanges` is not empty, and `isDirty` is false (this is an undo operation; therefore, after applying the changes, `modified` needs to be set manually).